### ISSUE TITLE
lsp-ui sideline not playing nice with repeated kill-line

### ISSUE THREAD

**Issue by antifuchs** [emacs-lsp/lsp-ui#244]
When using lsp-ui, having the sideline visible and using kill-line repeatedly (to delete a few successive lines interactively) breaks editing pretty terribly.

I am reproducing this in a go-mode buffer, but have seen similar behavior in other modes also. Example time - I'm using a function from [veneur's server.go](https://github.com/stripe/veneur/blob/master/server.go#L635), running `bingo` as the lsp server:

``` go
	go func() {
		log.Info("Starting Event worker")
              // ^ (point) is at the start of `log` on the line above
		defer func() {
			ConsumePanic(s.Sentry, s.TraceClient, s.Hostname, recover())
		}()
		s.EventWorker.Work()
	}()
```

Place point at the beginning of `log`, wait for the sideline to pop up. Press `C-k` twice, and everything looks mostly fine so far. Press it one more time, and (point) will jump all the way to the right (to after the sideline), and subsequent presses of `C-k` will super mess up the display of the sideline (they'll delete the right code, but there will be leftover whitespace, and the sideline jumps to be inline).

This formatting mess is not terribly problematic in go, as I have gofmt, but is still a little frustrating to deal with, as it messes with my feel for what I'm deleting.

As for expected behavior, I think what I'd like is for larger-scale delete operations on a line, sideline should hide itself, as its information would no longer realistically apply to the line, I think.

This may be related to/fixable by #187, I'm not sure!

**Comment by seagle0128** [comment]
Same issue. That's why I have to disable sideline.

**Comment by sebastiencs** [comment]
It should be fixed with 6041082a3873f3a22ccaf25dc8fbb4ce2bc9fbf0

**Comment by antifuchs** [comment]
Confirmed! This works for me in latest master, containing 6041082.
